homeassistant/components/mopar/__init__.py

```diff
@@ -16,34 +25,76 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-ATTR_VEHICLE_INDEX = 'vehicle_index'
 COOKIE_FILE = 'mopar_cookies.pickle'
 SUCCESS_RESPONSE = 'completed'
 
-MIN_TIME_BETWEEN_UPDATES = timedelta(days=7)
+SUPPORTED_PLATFORMS = [LOCK, SENSOR, SWITCH]
+
+DEFAULT_INTERVAL = timedelta(days=7)
 
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
         vol.Required(CONF_USERNAME): cv.string,
         vol.Required(CONF_PASSWORD): cv.string,
         vol.Required(CONF_PIN): cv.positive_int,
+        vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_INTERVAL):
+            vol.All(cv.time_period, cv.positive_timedelta),
     })
 }, extra=vol.ALLOW_EXTRA)
 
+SERVICE_HORN = 'sound_horn'
+ATTR_VEHICLE_INDEX = 'vehicle_index'
+SERVICE_HORN_SCHEMA = vol.Schema({
+    vol.Required(ATTR_VEHICLE_INDEX): cv.positive_int
+})
+
 
 def setup(hass, config):
     """Set up the Mopar component."""
     import motorparts
+
     cookie = hass.config.path(COOKIE_FILE)
     try:
         session = motorparts.get_session(
-            config.get(CONF_USERNAME), config.get(CONF_PASSWORD),
-            config.get(CONF_PIN), cookie_path=cookie)
+            config[CONF_USERNAME],
+            config[CONF_PASSWORD],
+            config[CONF_PIN],
+            cookie_path=cookie
+        )
     except motorparts.MoparError:
         _LOGGER.error("Failed to login")
         return False
 
-    hass.data[DOMAIN] = MoparData(hass, session)
+    data = hass.data[DOMAIN] = MoparData(hass, session)
+    data.update()
+
+    track_time_interval(
+        hass, data.update, config[CONF_SCAN_INTERVAL]
+    )
+
+    def handle_horn(call):
+        """Enable the horn on the Mopar vehicle."""
+        try:
+            motorparts.remote_command(
+                data.session,
+                motorparts.COMMAND_HORN,
+                call.data[ATTR_VEHICLE_INDEX]
+            )
+        except motorparts.MoparError as error:
+            _LOGGER.error(error)
+
+    hass.services.register(
+        DOMAIN,
+        SERVICE_HORN,
+        handle_horn,
+        schema=SERVICE_HORN_SCHEMA
+    )
+
+    for platform in SUPPORTED_PLATFORMS:
+        hass.async_create_task(
+            async_load_platform(hass, platform, DOMAIN, {}, config)
```

Comment:
We should use the sync api here since we're in sync context in `setup`.
```py
load_platform(...)
```

Reply:
Doesn't the fact that we are calling create_task mean that the task is running in the async context?

Reply:
Yeah, but we can't call any async functions, like `hass.async_create_task`, from sync context.

Reply:
Can I call `hass.add_job` or do I need to directly just call `load_platform` and not do it within the event loop?

Reply:
Just call `load_platform`.